packages/hash/frontend/src/blocks/page/usePortals.tsx

```diff
@@ -50,11 +53,26 @@ export const usePortals = () => {
     setPortals(blankPortals);
   }, []);
 
-  const renderedPortals = Array.from(portals.entries()).map(
-    ([target, { key, reactNode }]) => (
-      <Fragment key={key}>{createPortal(reactNode, target)}</Fragment>
-    ),
+  const groupedPortals = Array.from(portals.entries()).reduce(
+    (obj, portal) => {
+      const id = portal[1].id;
+
+      return { ...obj, [id]: [...(obj[id] || []), portal] };
+    },
+    {} as {
+      [id: string]: Array<[HTMLElement, BlockPortal]>;
```

Comment:
We probably ought to set up a linter around this

```suggestion
      [id: string]: [HTMLElement, BlockPortal][];
```